### ISSUE TITLE
`max-empty-lines` reads the byte-order mark as a character of the file, so a leading empty line behind it goes uncounted and every position stands a column right

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#601]
The check of this rule counts the breaks of `root.toString()`, and PostCSS puts the byte-order mark the file opened with back into that text while keeping it out of `root.source.input.css`, which every position is resolved against. So behind a mark the rule reads a file one character longer than Stylelint does, and two things go wrong with it.

A leading empty line goes uncounted. The first break of the text stands at index 1 rather than 0, so the check for the beginning of the file, `!matchStartIndex`, does not fire for it, and the run the file opens with counts one line short: `"﻿\n\na {}"` is silent under a maximum of one, where `"\n\na {}"` draws a warning. Every position behind the mark is one column right, since the index the warning is reported at counts the mark and the text the column is looked up in does not: `"﻿\n\na {}"` under a maximum of zero reports `3:1` — the line of `a {}` — for the break that opens line 2, and `"﻿a {}\n\n\n"` under one reports `4:1` and `4:2`, a second column on an empty line.

## The runs

On `28fe7bc` and on `f17bb17` alike, under `css`:

```
"﻿\n\na {}"      max 1 → no warning              "\n\na {}"   max 1 → 2:1
"﻿\n\n\na {}"    max 1 → 4:1                     "\n\n\na {}" max 1 → 2:1, 3:1
"﻿\n\na {}"      max 0 → 3:1                     "\n\na {}"   max 0 → 1:1, 2:1
"﻿a {}\n\n\n"    max 1 → 4:1, 4:2                "a {}\n\n\n" max 1 → 3:1, 4:1
```

The fix writes the raws, which never hold the mark, so what it writes is right where it writes at all; a stylesheet holding a mark and nothing but breaks is the one place the two readings meet, and the branch closing #404 writes its run as the run the file opens with while the check still counts it as one that does not.

## What would end it

Counting over the text the positions are resolved against, `root.source.input.css`, or taking the mark's length off every index where the file opened with one.
